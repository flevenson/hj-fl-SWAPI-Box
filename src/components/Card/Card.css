.card {
	border: solid;
}

.favorite-button {
	cursor: pointer;
	height: 20px;
	width: 20px;
}