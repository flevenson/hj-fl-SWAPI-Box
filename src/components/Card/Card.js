import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ name, gender, height }) => {
	// const { name, gender, height } = person;

	  return (
	    <div>
	      <p>{name}</p>
	      <p>{gender}</p>
	      <p>{height}</p>
	    </div>
	  )

	}

Card.propTypes = {
	name: PropTypes.string,
	gender: PropTypes.string,
	height: PropTypes.number
}

export default Card; 