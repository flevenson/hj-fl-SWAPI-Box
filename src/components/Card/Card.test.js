import React from 'react';
import { shallow } from 'enzyme';

import Card from './Card.js';

describe('Card', () => {

	it('should match a snapshot', () => {
		const mockData = [{
			name: 'Joe',
			gender: 'male',
			height: 165  
		}]

		const wrapper = shallow(<Card person={mockData} />)
		expect(wrapper).toMatchSnapshot()
	})
})