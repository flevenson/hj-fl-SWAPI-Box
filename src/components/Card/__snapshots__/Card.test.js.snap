// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={Array []}
    selected=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <button
          onClick={[Function]}
        >
          <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />
        </button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />,
          "name": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "favorite button",
            "className": "favorite-button",
            "name": undefined,
            "src": "star.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <button
            onClick={[Function]}
          >
            <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />
          </button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />,
            "name": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "favorite button",
              "className": "favorite-button",
              "name": undefined,
              "src": "star.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should match the snapshot after people has been selected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
      Array [
        Object {
          "homeworld": "Earth",
          "name": "Joe",
          "population": 5,
          "species": "human",
        },
      ]
    }
    selected="people"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <p>
            0
            : 
            Object {
              "homeworld": "Earth",
              "name": "Joe",
              "population": 5,
              "species": "human",
            }
          </p>,
        ],
        <button
          onClick={[Function]}
        >
          <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />
        </button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "0",
            ": ",
            Object {
              "homeworld": "Earth",
              "name": "Joe",
              "population": 5,
              "species": "human",
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          "0",
          ": ",
          Object {
            "homeworld": "Earth",
            "name": "Joe",
            "population": 5,
            "species": "human",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />,
          "name": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "favorite button",
            "className": "favorite-button",
            "name": undefined,
            "src": "star.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <p>
              0
              : 
              Object {
                "homeworld": "Earth",
                "name": "Joe",
                "population": 5,
                "species": "human",
              }
            </p>,
          ],
          <button
            onClick={[Function]}
          >
            <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />
          </button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "0",
              ": ",
              Object {
                "homeworld": "Earth",
                "name": "Joe",
                "population": 5,
                "species": "human",
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            "0",
            ": ",
            Object {
              "homeworld": "Earth",
              "name": "Joe",
              "population": 5,
              "species": "human",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />,
            "name": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "favorite button",
              "className": "favorite-button",
              "name": undefined,
              "src": "star.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should match the snapshot after planets has been selected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
      Array [
        Object {
          "climate": "Hot",
          "name": "Earth",
          "population": 5,
          "residents": Array [
            "Luke",
            "Yoda",
            "Annikan",
          ],
          "terrain": "mountainous",
        },
      ]
    }
    selected="planets"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <p>
            0
            : 
            Object {
              "climate": "Hot",
              "name": "Earth",
              "population": 5,
              "residents": Array [
                "Luke",
                "Yoda",
                "Annikan",
              ],
              "terrain": "mountainous",
            }
          </p>,
        ],
        <button
          onClick={[Function]}
        >
          <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />
        </button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "0",
            ": ",
            Object {
              "climate": "Hot",
              "name": "Earth",
              "population": 5,
              "residents": Array [
                "Luke",
                "Yoda",
                "Annikan",
              ],
              "terrain": "mountainous",
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          "0",
          ": ",
          Object {
            "climate": "Hot",
            "name": "Earth",
            "population": 5,
            "residents": Array [
              "Luke",
              "Yoda",
              "Annikan",
            ],
            "terrain": "mountainous",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />,
          "name": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "favorite button",
            "className": "favorite-button",
            "name": undefined,
            "src": "star.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <p>
              0
              : 
              Object {
                "climate": "Hot",
                "name": "Earth",
                "population": 5,
                "residents": Array [
                  "Luke",
                  "Yoda",
                  "Annikan",
                ],
                "terrain": "mountainous",
              }
            </p>,
          ],
          <button
            onClick={[Function]}
          >
            <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />
          </button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "0",
              ": ",
              Object {
                "climate": "Hot",
                "name": "Earth",
                "population": 5,
                "residents": Array [
                  "Luke",
                  "Yoda",
                  "Annikan",
                ],
                "terrain": "mountainous",
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            "0",
            ": ",
            Object {
              "climate": "Hot",
              "name": "Earth",
              "population": 5,
              "residents": Array [
                "Luke",
                "Yoda",
                "Annikan",
              ],
              "terrain": "mountainous",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />,
            "name": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "favorite button",
              "className": "favorite-button",
              "name": undefined,
              "src": "star.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should match the snapshot after vehicles has been selected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
      Array [
        Object {
          "class": "Rocket",
          "max_atmosphering_speed": 500,
          "model": "3549",
          "name": "Spaceship",
          "passengers": 5,
        },
      ]
    }
    selected="vehicles"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <p>
            0
            : 
            Object {
              "class": "Rocket",
              "max_atmosphering_speed": 500,
              "model": "3549",
              "name": "Spaceship",
              "passengers": 5,
            }
          </p>,
        ],
        <button
          onClick={[Function]}
        >
          <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />
        </button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "0",
            ": ",
            Object {
              "class": "Rocket",
              "max_atmosphering_speed": 500,
              "model": "3549",
              "name": "Spaceship",
              "passengers": 5,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          "0",
          ": ",
          Object {
            "class": "Rocket",
            "max_atmosphering_speed": 500,
            "model": "3549",
            "name": "Spaceship",
            "passengers": 5,
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="favorite button"
            className="favorite-button"
            src="star.svg"
          />,
          "name": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "favorite button",
            "className": "favorite-button",
            "name": undefined,
            "src": "star.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <p>
              0
              : 
              Object {
                "class": "Rocket",
                "max_atmosphering_speed": 500,
                "model": "3549",
                "name": "Spaceship",
                "passengers": 5,
              }
            </p>,
          ],
          <button
            onClick={[Function]}
          >
            <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />
          </button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "0",
              ": ",
              Object {
                "class": "Rocket",
                "max_atmosphering_speed": 500,
                "model": "3549",
                "name": "Spaceship",
                "passengers": 5,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            "0",
            ": ",
            Object {
              "class": "Rocket",
              "max_atmosphering_speed": 500,
              "model": "3549",
              "name": "Spaceship",
              "passengers": 5,
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="favorite button"
              className="favorite-button"
              src="star.svg"
            />,
            "name": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "favorite button",
              "className": "favorite-button",
              "name": undefined,
              "src": "star.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
